commands/cli.js

```diff
@@ -66,17 +51,58 @@ export async function estimateTxsFees(archethic, transactions) {
   return { refTxFees, filesTxFees }
 }
 
-function handleDirectory(entry, files) {
-  if (fs.statSync(entry).isDirectory()) {
-    fs.readdirSync(entry).forEach(child => {
-      handleDirectory(entry + path.sep + child, files)
-    });
+export function getFiles(folderPath, includeGitIgnoredFiles = false) {
+  let files = []
+  if (fs.statSync(folderPath).isDirectory()) {
+    handleDirectory(folderPath, files, includeGitIgnoredFiles)
+
+    files = files.map((file) => {
+      file.filePath = file.filePath.replace(folderPath, '')
+      return file
+    })
   } else {
-    handleFile(entry, files)
+    const data = fs.readFileSync(folderPath)
+    const file_path = PathLib.basename(folderPath)
+    files.push({ "filePath": file_path, data })
   }
+
+  return files
 }
 
-function handleFile(filePath, files) {
-  const data = fs.readFileSync(filePath)
-  files.push({ filePath, data })
+function handleDirectory(entry, files, includeGitIgnoredFiles) {
+  let filters = []
+
+  if (!includeGitIgnoredFiles) {
+    let gitIgnoreFilePath = PathLib.join(entry, '.gitignore')
+    console.log(gitIgnoreFilePath)
+
+    if (fs.existsSync(gitIgnoreFilePath)) {
+      filters = parse(fs.readFileSync(gitIgnoreFilePath))['patterns']
+    }
+  }
+  filters.unshift('.gitignore')
+  filters.unshift('.git')
+
+  const isGitIgnored = ignore().add(filters)
+  doHandleDirectory(entry, files, isGitIgnored)
+}
+
+function doHandleDirectory(entry, files, isGitIgnored) {
+  // reduce search space by omitting folders at once
+  if (fs.statSync(entry).isDirectory() && !isGitIgnored.ignores(entry)) {
+    fs.readdirSync(entry).forEach((child) => {
+      doHandleDirectory(entry + PathLib.sep + child, files, isGitIgnored)
```

Comment:
you are using files (`entry + PathLib.sep + child`) on a function called `doHandleDirectory`, it's very confusing. I suggest renaming the `entry` to `folderPath` for explicitness.

Then I suggest not to call that recursion with files.

```javascript
function doHandleDirectory(folderPath, files, isGitIgnored) {
  // reduce search space by omitting folders at once
  if (!isGitIgnored.ignores(folderPath)) {
    fs.readdirSync(folderPath).forEach((fileOrFolderPath) => {
      if (fs.statSync(fileOrFolderPath).isDirectory()) {
        doHandleDirectory(fileOrFolderPath, files, isGitIgnored)
      } else {
        doHandleFile(filePath, files, isGitIgnored); 
      }
    })
  } 
}
```